### PR TITLE
Use translation api to translate comments

### PR DESCRIPTION
**Pull request by ALai2**
Used the translation api to translate comments in the server. It checks to see if the language code is valid, in other words the option is to translate and not leave the original message as is, and then translates the comment and sends it back to script.js as a json. I was able to test the code by deploying it.
https://alisal-step-2020.appspot.com/comments.html

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -41,20 +45,30 @@
...
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<Entity> queryList = results.asList(FetchOptions.Builder.withLimit(commentLimit));
 
+    Translate translate = null;
+    if (!languageCode.equals("original")) {
+      translate = TranslateOptions.getDefaultInstance().getService();
+    }
```
Comment (linxiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You could directly use
"Translate translate = TranslateOptions.getDefaultInstance().getService();"

You will overwrite the value anyway later when the language code is not original.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -41,20 +45,30 @@
...
     ArrayList<Comment> commentList = new ArrayList<Comment>();
     for (Entity entity : queryList) {
       long id = entity.getKey().getId();
       String name = (String) entity.getProperty("name");
       String comment = (String) entity.getProperty("comment");
       long timestamp = (long) entity.getProperty("timestamp");
 
+      // Do the translation.
+      if (!languageCode.equals("original")) {
+        Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage(languageCode));
```
Comment (linxiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Do you want to add error handling for TranslateException?

Comment (ALai2) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I added a try catch block which handles error handling for TranslateException. In the catch block I made a println statement which prints the error in the terminal, but should the user be able to see that the error occurred? I was thinking about making a variable called error inside the comment class to be displayed using javascript when the json is returned. What do you think about that?

Comment (ALai2) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: So I made a commit which catches the error and sends it back to script.js as a response which script.js would then show as an alert.

Comment (linxiz) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Usually user see some error message to know there is an error but without detailed error message which is related to detailed server implementation. Server could have server logs for debugging on server side with more detailed info.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -41,28 +47,47 @@
...
-    String json = gson.toJson(commentList);
-    response.setContentType("application/json; charset=utf-8");
-    response.getWriter().println(json);
+      Gson gson = new Gson();   
+      String json = gson.toJson(commentList);
+      response.setContentType("application/json; charset=utf-8");
+      response.getWriter().println(json);
+
+    } catch (TranslateException error) {
+      String errorMessage = "Google Translate API returned an error " + error.getMessage();
```
Comment (linxiz) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: usually error msg shown to the user could be something like "error translating", or "error translating message: <input msg>", you may not want to expose which api you use on server side to the user

**Review by linxiz — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (ALai2) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: That makes sense. I added a commit which shortens the error message shown to the user as simply "Error translating message."

**Pull request merged**